test/integration/crud/bulk.test.js

```diff
@@ -22,6 +27,74 @@ describe('Bulk', function () {
   before(function () {
     return setupDatabase(this.configuration);
   });
+  describe('BulkOperationBase', () => {
+    describe('#raw', function () {
+      let client;
+      beforeEach(async function () {
+        client = this.configuration.newClient();
+        await client.connect();
+      });
+      afterEach(async function () {
+        await client.close();
+      });
+      context('when called with an undefined operation', function () {
+        it('should throw a MongoInvalidArgument error ', async function () {
+          const bulkOp = client.db('test').collection('test').initializeUnorderedBulkOp();
+          expect(() => bulkOp.raw(undefined)).to.throw(MongoInvalidArgumentError);
+          expect(() => bulkOp.raw(true)).to.throw(MongoInvalidArgumentError);
+          expect(() => bulkOp.raw(3)).to.throw(MongoInvalidArgumentError);
+        });
+      });
+
+      context('when called with a valid operation', function () {
+        it('should not throw a MongoInvalidArgument error', async function () {
+          try {
+            client.db('test').collection('test').initializeUnorderedBulkOp().raw({ insertOne: {} });
+          } catch (error) {
+            expect(error).not.to.exist;
+          }
+        });
+      });
+    });
+  });
+
+  describe('Db.collection', function () {
+    describe('#insertMany', function () {
+      let client;
+      beforeEach(async function () {
+        client = this.configuration.newClient();
+        await client.connect();
+      });
+      afterEach(async function () {
+        await client.close();
+      });
+      context('when passed an invalid or sparse list', function () {
+        it('insertMany should throw a MongoInvalidArgument error when called with a valid operation', async function () {
```

Comment:
I think this should say "invalid', but, in particular, we want to be specific that we are testing a sparse array in this example. I'm not 100% sure how helpful the actual thrown error would be to the user in this case, since they would get "Operation must be an object with an operation key" - which doesn't necessarily tell them what is wrong with the top level argument they passed into insertMany.